mlir/include/mlir/Dialect/OpenMP/OpenMPOps.td

```diff
@@ -565,6 +570,10 @@ def CanonicalLoopOp : OpenMP_Op<"canonical_loop", [SameVariadicOperandSize,
 
     /// Returns the induction variables of the canonical loop nest.
     ArrayRef<BlockArgument> getIVs() { return getRegion().getArguments(); }
+
+    /// Populates `wrappers` with the list of wrapper operations around this
+    /// canonical loop.
+    void getWrappers(llvm::SmallVectorImpl<mlir::Operation *> &wrappers);
```

Comment:
I was concerned with the potential for copies, but I don't think it makes sense given it will have few elements and they are going to be small as well. Done.

Reply:
@ergawy [LLVM's Programmers Manual](https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h) discourages the use of `SmallVector` in the API. It only mentions parameters explicitly but I would understand the return value as implicit out-parameter. There even is an example using `SmallVectorImpl` as out-parameter as "ENCOURANGED" instead of a return value. I think it is supposed to be the canonical coding style.

The reason I think is that the the small vector size should not be part of the the API (but the caller controls the appropriate small size), even if with NRO there should no additional copy be involved. The example on [std::vector](https://llvm.org/docs/ProgrammersManual.html#vector) even shows an optimization that is not possible when returning pre-allocated memory.